tt_metal/impl/program/program.hpp

```diff
@@ -205,16 +208,16 @@ class Program {
     friend std::shared_ptr<CircularBuffer> detail::GetCircularBuffer(const Program &program, CBHandle id);
     friend void detail::ValidateCircularBufferRegion(const Program &program, const Device *device);
 
-    friend KernelHandle detail::AddKernel(Program &program, std::shared_ptr<Kernel> kernel, const CoreType &core_type);
+    friend KernelHandle detail::AddKernel(Program &program, std::shared_ptr<Kernel> kernel, const HalProgrammableCoreType core_type);
     friend std::shared_ptr<Kernel> detail::GetKernel(const Program &program, KernelHandle kernel_id);
 
     friend uint32_t CreateSemaphore(Program &program, const std::variant<CoreRange,CoreRangeSet> &core_spec, uint32_t initial_value, CoreType core_type);
```

Comment:
Should CreateSemaphore API be updated to not expose CoreType as well?

Reply:
Actually, I realize this isn't the changes to move CoreType to ProgrammableCoreType. But this is nice to cleanup, I think CreateSemaphore is the only host api where CoreType is exposed now.

Reply:
I was thinking about this, not 100% clear to me yet.  It definitely would clean up a bunch of places to make this ProgrammableCoreType. At the meeting next week about virtual coords, I think we should discuss how we want this to go through the API: ProgrammableCoreType, CoreType or have coords include their type.

Reply:
this might get updated based on what we decide for laying out logical cores/grids so maybe hold off on the api change?